### ISSUE TITLE
Duplicate GUID (Delphi)

### ISSUE THREAD

**Issue by peardox** [castle-engine/castle-engine#517]
I'm just building an AllExamples project group and this popped up...

![dupeguid](https://github.com/castle-engine/castle-engine/assets/1549219/a108b259-e940-44a8-9e95-e822bb908788)

This also happened for fps_game_standalone

The project group was created by navigating to examples in Explorer then searching for *.dproj which creates an view of all trhe dproj files. These can then be all selected and simply dragged into a project after deleting the default one Delphi generates itself

145 dproj files... Wonder how long to build them all :)

**Comment by peardox** [comment]
FYI

[AllExamples.zip](https://github.com/castle-engine/castle-engine/files/11957391/AllExamples.zip)

Stick it in examples and unzip it (stupid Github don't like the groupproj ext)

**Comment by peardox** [comment]
Jeez - it's building CGE 145 times from the look of it :(

There's a prefectly good CGE already installed it's completely ignoring....

**Comment by michaliskambi** [comment]
Duplicate GUID is a consequence of our `--guid-from-name`, that we use only for examples, to have `examples\regenerate_auto_files_in_all_examples.sh` repeatable. So CGE examples with equal project names will indeed have colliding GUIDs.

It doesn't affect normal users, it doesn't affect normal projects (GUIDs are not deterministic based on name, normally).

In the long run, `examples\deprecated_to_upgrade\isometric_game\` and `examples\deprecated_to_upgrade\fps_game_old_ai\` should just be removed (both of these should happen before 7.0 release), so the problem will be solved this way. For now just ignore it.

AllExamples -- thank you, though if we decide to add it, we definitely need application to autogenerate this file. We don't want to have to maintain it each time we add new example :)

**Comment by peardox** [comment]
Yeah I noticed all the deprecated ones when weeding out the bad ones too. They should really either be removed or not included in AllExamples

I wrote code to auto generate the group project file for MakeP4DComponents although admitedly it didn't use a XML library - just  output the specific lines XML lib would have generated.

I have already thought of doing the same for CGE's examples. I thought that something similar to a gitignore could control a scan of dirs that looks jor dproj files and builds based on that.

One thing that would make it better is if the examples in AllExamples were grouped by subject / TLD as it'd make navigating the rather long list of projects much easier.

It would also be nice to have the ability to potentially re-write the dproj files on request to use an installed CGE (like the planned getit one) rather than re-building it each time
